### PR TITLE
Top three query

### LINKED ISSUE #29: create graphql query for currentContenders
- [ ] create graphql query for currentContenders to return info for top three rated verses; ordered by votes; grouped by month
- [ ] create type for currentContenders

**request:**
query currentContenders {
 id
 path
 title
 user { userType }
 votes { voteType }
}

**response**
{

 "data": [
 {
    "id": <integer>
    "path": <string>,
    "title": <string>,
    "description": <string>,
    "user": {
     "id": <integer>
        "name": <string>,
           "image": <string>
   }
          "votes": {
    "id": <integer>
    }
    }
 \\ 2 more verse hashes like this ^
 ]
}

### PR DESCRIPTION
**Pull request by the-color-bliu**
## Create query for top three contenders of the current competition
Create type, add method to query type, add method to verse model, create respective tests

This PR closes #29 

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [x] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [x] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: app/models/verse.rb [hunk 1]
```diff
@@ -13,11 +13,22 @@ def self.last_month_winner
...
-    current_month = Time.now.mon
     if current_month == 1
       return 12
     else
       current_month - 1
     end
   end
+
+  def self.current_month
+    current_month = Time.now.mon
```
Comment (SageOfCode) [inline app/models/verse.rb:24]: I don't think we need to assign the variable here since that is all the method is returning.

**Review by SageOfCode — APPROVED** [review]
Great work here, this looks great Brian!

File: app/models/verse.rb [hunk 2]
```diff
@@ -13,11 +13,22 @@ def self.last_month_winner
...
+  end
+
+  def self.top_three
+    competition = Competition.find_by(month: current_month)
+    contenders = competition.verses.sort_by do |verse|
+      verse.votes.count
+    end.reverse
+    contenders[0..2]
```
Comment (ETBassist) [inline app/models/verse.rb:32]: This looks good, though I think we want to refactor this into ActiveRecord at some point for teh performance;  made an issue as a reminder

File: spec/requests/queries/top_three_contenders_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    top_three = result[:data][:topThreeContenders]
+    expect(top_three).to be_an(Array)
+    expect(top_three[0]).to have_key(:audioPath)
+    expect(top_three[0][:audioPath]).to eq(verse3.audio_path)
+    expect(top_three[0]).to have_key(:title)
+    expect(top_three[0][:title]).to eq(verse3.title)
+    expect(top_three[0]).to have_key(:user)
+    expect(top_three[0][:user]).to be_a(Hash)
+    expect(top_three[0][:user]).to have_key(:name).and have_key(:image)
```
Comment (ETBassist) [inline spec/requests/queries/top_three_contenders_spec.rb:44]: Oooooh, didn't realize you could chain expectations like that

**Review by ETBassist — APPROVED** [review]
Jackie Chan Approves

File: spec/models/verse_spec.rb [hunk 4]
```diff
@@ -10,20 +10,49 @@
...
+      verse10 = create(:verse, user_id: @user5.id, competition_id: competition2.id)
+
+      vote7 = Vote.create(verse_id: verse7.id, user_id: @user1.id)
+      vote8 = Vote.create(verse_id: verse8.id, user_id: @user1.id)
+      vote9 = Vote.create(verse_id: verse8.id, user_id: @user2.id)
+      vote10 = Vote.create(verse_id: verse6.id, user_id: @user3.id)
+      vote11 = Vote.create(verse_id: verse8.id, user_id: @user4.id)
+      vote12 = Vote.create(verse_id: verse7.id, user_id: @user5.id)
+      
+      expect(Verse.top_three).to eq([verse8, verse7, verse6])
```
Comment (nmking22) [inline spec/models/verse_spec.rb:55]: Nice use of TDD here!

File: app/models/verse.rb (hunk 2 above)
Comment (nmking22) [reply app/models/verse.rb:32]: Good call Eugene - an order method here should do the trick

**Review by nmking22 — APPROVED** [review]

**Pull request merged**